Code/Framework/AzToolsFramework/Tests/Prefab/PrefabFocus/PrefabFocusTests.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/EntityId.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/Entity/PrefabEditorEntityOwnershipInterface.h>
+#include <AzToolsFramework/Prefab/PrefabFocusInterface.h>
+#include <Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    class PrefabFocusTests
+        : public PrefabTestFixture
+    {
+    protected:
+        void GenerateTestHierarchy()
+        {
+            /*
+             *  City (Prefab Container)
+             *  |_  City
+             *  |_  Street (Prefab Container)
+             *      |_  Car (Prefab Container)
+             *      |   |_ Passenger
+             *      |_  SportsCar (Prefab Container)
+             *          |_ Passenger
+             */
+
+            m_entityMap["passenger1"] = CreateEntity("Passenger1");
+            m_entityMap["passenger2"] = CreateEntity("Passenger2");
+            m_entityMap["city"] = CreateEntity("City");
+
+            AzToolsFramework::EditorEntityContextRequestBus::Broadcast(
+                &AzToolsFramework::EditorEntityContextRequests::HandleEntitiesAdded,
+                AzToolsFramework::EntityList{ m_entityMap["passenger1"], m_entityMap["passenger2"], m_entityMap["city"] });
+
+            AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> carInstance =
+                m_prefabSystemComponent->CreatePrefab({ m_entityMap["passenger1"] }, {}, "test/car");
+            ASSERT_TRUE(carInstance);
+            m_instanceMap["car"] = carInstance.get();
+
+            AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> sportsCarInstance =
+                m_prefabSystemComponent->CreatePrefab({ m_entityMap["passenger2"] }, {}, "test/sportsCar");
+            ASSERT_TRUE(sportsCarInstance);
+            m_instanceMap["sportsCar"] = sportsCarInstance.get();
+
+            AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> streetInstance =
+                m_prefabSystemComponent->CreatePrefab({}, MakeInstanceList( AZStd::move(carInstance), AZStd::move(sportsCarInstance) ), "test/street");
+            ASSERT_TRUE(streetInstance);
+            m_instanceMap["street"] = streetInstance.get();
+
+            m_rootInstance =
+                m_prefabSystemComponent->CreatePrefab({ m_entityMap["city"] }, MakeInstanceList(AZStd::move(streetInstance)), "test/city");
+            ASSERT_TRUE(m_rootInstance);
+            m_instanceMap["city"] = m_rootInstance.get();
+        }
+
+        AZStd::unordered_map<AZStd::string, AZ::Entity*> m_entityMap;
+        AZStd::unordered_map<AZStd::string, Instance*> m_instanceMap;
+
+        AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> m_rootInstance;
+    };
+
+    TEST_F(PrefabFocusTests, PrefabFocus_FocusOnOwningPrefab)
```

Comment:
suggestion: This is okay for now but again this one test case is testing a lot of different things. Having smaller tests testing one thing at a time makes diagnosing failures much easier, I'll ping you a link offline